src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    public void growIfArrayIsFull() {
+        if (size == elements.length) {
+            float multipler = 1.5f;
+            int newCapacity = Math.round(elements.length * multipler);
+            System.arraycopy(elements, 0, elements = new Object[newCapacity], 0, size);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayIsFull();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
```

Comment:
The index check logic is repeated in the add(), get(), set(), and remove() methods. Consider moving it to a separate private method to avoid code duplication.